xgi/generators/uniform.py

```diff
@@ -299,28 +308,42 @@ def uniform_erdos_renyi_hypergraph(n, m, p, p_type="degree", seed=None):
     H.add_nodes_from(node_labels)
 
     if p_type == "degree":
-        q = p / (m * n ** (m - 1))  # wiring probability
+        if multiedges:
+            q = p / (m * n ** (m - 1))  # wiring probability
+        else:
+            q = p * n / (m * comb(n, m))
     elif p_type == "prob":
         q = p
     else:
         raise XGIError("Invalid p_type!")
 
     if q > 1 or q < 0:
-        raise XGIError("Probability not in [0,1].")
+        raise XGIError("Probability not in [0, 1].")
+
+    if multiedges:
+        max_index = n**m
+        index = np.random.geometric(q) - 1  # -1 b/c zero indexing
+        f = _index_to_edge_prod
+    else:
+        max_index = comb(n, m, exact=True)
+        index = np.random.geometric(q)
```

Comment:
why not -1 here like in multiedge case?

Reply:
I fixed it so they're both zero-indexed.